src/bindings/python/src/pyopenvino/utils/utils.hpp

```diff
@@ -32,6 +32,28 @@
 
 namespace py = pybind11;
 
+// Conditional GIL management for PEP 703 (free-threaded Python)
+#if defined(Py_GIL_DISABLED) && Py_GIL_DISABLED
```

Comment:
Consider replace it to compile time flag and use more C++ instead macros?
```suggestion
inline constexpr bool PY_GIL_DISABLED {
#if defined(Py_GIL_DISABLED) && Py_GIL_DISABLED
true
#endif
};
```

Reply:
done